test/new-e2e/tests/apm/tests.go

```diff
@@ -93,6 +93,27 @@ func testAutoVersionTraces(t *testing.T, c *assert.CollectT, intake *components.
 	}
 }
 
+func tracesSampledByProbabilitySampler(t *testing.T, c *assert.CollectT, intake *components.FakeIntake) {
+	t.Helper()
+	traces, err := intake.Client().GetTraces()
+	assert.NoError(c, err)
+	assert.NotEmpty(c, traces)
+	t.Log("Got traces", traces)
+	for _, p := range traces {
+		for _, tp := range p.AgentPayload.TracerPayloads {
+			for _, chunk := range tp.Chunks {
+				dm, ok := chunk.Tags["_dd.p.dm"]
+				if !ok {
+					t.Errorf("Expected trace chunk tags to contain _dd.p.dm, but it does not.")
+				}
+				if dm != "-9" {
+					t.Errorf("Expected dm == -9, but got %v", dm)
+				}
```

Comment:
Yes, it's always difficult to do e2e stuff on probabilistic things. We either end up with flaky tests or being too broad in the variance we accept, or both.

It would be nice to be able to generate specific trace IDs with the generator so we can know precisely which traces should be sent through.